src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Deployments/GetAzureDeploymentCmdlet.cs

```diff
@@ -0,0 +1,60 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Components;
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels;
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
+{
+    /// <summary>
+    /// Get deployments.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "AzureRmDeployment", DefaultParameterSetName = GetAzureDeploymentCmdlet.DeploymentNameParameterSet), OutputType(typeof(PSDeployment))]
+    public class GetAzureDeploymentCmdlet : ResourceManagerCmdletBase
+    {
+        /// <summary>
+        /// The deployment Id parameter set.
+        /// </summary>
+        internal const string DeploymentIdParameterSet = "GetByDeploymentId";
+
+        /// <summary>
+        /// The deployment name parameter set.
+        /// </summary>
+        internal const string DeploymentNameParameterSet = "GetByDeploymentName";
+
+        [Alias("DeploymentName")]
+        [Parameter(Position = 0, ParameterSetName = GetAzureDeploymentCmdlet.DeploymentNameParameterSet, Mandatory = false,
+            ValueFromPipelineByPropertyName = true, HelpMessage = "The name of deployment.")]
```

Comment:
@Tiano2017 please remove the `ValueFromPipelineByPropertyName` property from the `-Name` parameter and move it to the `-ResourceId` parameter so users can do the following:

```powershell
Get-AzureRmResource -ResourceType Microsoft.Resources/deployments | Get-AzureRmDeployment
```

Reply:
we need it on the Name parameter for some other pipeline scenarios. As for Get-AzureRmResource, I don't think it returns "deployments" as resources. Strictly speaking, "deployment" is not a resource, although we implement the APIs to be consistent with resource APIs.

Reply:
@Tiano2017 this is deprecated - Piping by value is now the recommendation - if you can pipe the object associated with the cmdlet noun into the cmdlet no by-property-name piping (other than piping resourceid by property name)  should be used.

Reply:
Sorry Mark.. I'm not quite getting the point. Are you recommending just removing this piping support? 

---
In reply to: [207954017](https://github.com/Azure/azure-powershell/pull/6516#discussion_r207954017) [](ancestors = 207954017)

Reply:
As replied in the other comment, Get-AzureRmResource doesn't return deployments, as deployment is not a resource..

---
In reply to: [207688618](https://github.com/Azure/azure-powershell/pull/6516#discussion_r207688618) [](ancestors = 207688618)

Reply:
removed. 

---
In reply to: [208426878](https://github.com/Azure/azure-powershell/pull/6516#discussion_r208426878) [](ancestors = 208426878)

Reply:
We have these two parameters for Get-AzureRmResourceGroupDeployment. I think we can keep them to be consistent. 

---
In reply to: [208427148](https://github.com/Azure/azure-powershell/pull/6516#discussion_r208427148) [](ancestors = 208427148)